spring-integration-file/src/main/java/org/springframework/integration/file/NewFileCallback.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2002-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.file;
+
+import java.io.File;
+
+/**
+ * A callback for newly created files in {@link FileWritingMessageHandler}.
+ *
+ * @author Alen Turkovic
+ */
+public interface NewFileCallback {
```

Comment:
Why can't we just stick with the `java.util.function.Consumer`?
I don't see reason in a new functional interface...

Reply:
No problem, will change. I was actually thinking of a BiConsumer<File, Message<?>> with the incoming message in case some information (for an example, a CSV header) is written in the message.

Reply:
Yes, that good idea, indeed!